drv/transceivers-server/src/main.rs

```diff
@@ -406,21 +418,55 @@ impl ServerImpl {
                 // be transient (and we'll remove the transceiver on the
                 // next pass through this function).
                 Err(FpgaError::ImplError(e)) => {
-                    match Reg::QSFP::PORT0_STATUS::Encoded::from_u8(e) {
+                    use Reg::QSFP::PORT0_STATUS::Encoded;
+                    match Encoded::from_u8(e) {
                         Some(val) => {
+                            if matches!(val, Encoded::I2cAddressNack) {
+                                self.consecutive_nacks[i] += 1;
+                                if self.consecutive_nacks[i] == 3 {
+                                    to_disable.set(port);
+                                }
+                            } else {
+                                self.consecutive_nacks[i] = 0;
+                            }
                             ringbuf_entry!(Trace::TemperatureReadError(i, val))
                         }
                         None => {
                             // Error code cannot be decoded
-                            ringbuf_entry!(Trace::InvalidPortStatusError(i, e))
+                            ringbuf_entry!(Trace::InvalidPortStatusError(i, e));
+                            self.consecutive_nacks[i] = 0;
                         }
                     }
                 }
                 Err(e) => {
+                    self.consecutive_nacks[i] = 0;
                     ringbuf_entry!(Trace::TemperatureReadUnexpectedError(i, e));
                 }
             }
         }
+        if !to_disable.is_empty() {
+            self.disable_ports(to_disable);
+        }
+    }
+
+    fn disable_ports(&mut self, mask: LogicalPortMask) {
+        for port in mask.to_indices() {
+            self.thermal_models[port.0 as usize] = None;
+        }
+        for (step, f) in [
+            Transceivers::assert_reset,
+            Transceivers::assert_lpmode,
```

Comment:
While asserting LPMode makes sense, in reality driving LPMode to some modules while they're unpowered causes other 3V3 issues, so we want this step to be `deassert_lpmode`. Our investigations of this on the board: https://github.com/oxidecomputer/hardware-qsfp-x32/issues/47#issuecomment-1329846157

Reply:
thanks, I remembered the leakage issue but got the sign wrong here!